### PR TITLE
[Fix #3782] Properly resolving root dir in all CI environments

### LINKED ISSUE #3782: Wrong value for project.root.dir property
### Describe the bug

`project.root.dir` property, defined in parent pom, relies on `maven.multiModuleProjectDirectory`, which is not documented and might fail as explained [here](https://stackoverflow.com/questions/29778262/what-is-maven-multimoduleprojectdirectory-used-for).
It should be replaced by proper usage of [ Maven directory plugin](https://github.com/jdcasey/directory-maven-plugin?tab=readme-ov-file#highest-basedir-goal)

### Expected behavior

_No response_

### Actual behavior

In any enviroment where test are not invoked from kogito-runtimes root, this causes test errors in modules using this property,
For example
```
[ERROR] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.061 s <<< FAILURE! -- in org.kie.kogito.quarkus.workflows.CallbackStateIT
[ERROR] org.kie.kogito.quarkus.workflows.CallbackStateIT.callbackStateWithError -- Time elapsed: 0.039 s <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <201> but was <404>.
``` 
In working scenarios, The callback json file is copied into the Integration test module from the Workflow builder module. When `project.root.dir `is wrong, the file is not found while copying,  hence is not there when the test is run, causing the not found failure. 

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Kogito version or git rev (or at least Quarkus version if you are using Kogito via Quarkus platform BOM)

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by fjtirado**
Fix https://github.com/apache/incubator-kie-kogito-runtimes/issues/3782
Merge with https://github.com/apache/incubator-kie-kogito-apps/pull/2144

### CODE REVIEW

File: kogito-build/kogito-build-no-bom-parent/pom.xml [hunk 1]
```diff
@@ -671,6 +670,23 @@
             <legacyMode>true</legacyMode>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.commonjava.maven.plugins</groupId>
+          <artifactId>directory-maven-plugin</artifactId>
+          <version>0.1</version>
```
Comment (gmunozfe) [inline kogito-build/kogito-build-no-bom-parent/pom.xml:676]: why not to use latest one?
```suggestion
          <version>1.0</version>
```

**Review by gmunozfe — APPROVED** [review]
Works perfectly, good catch @fjtirado !
Returns root dir from any child module, checked with `mvn directory:highest-basedir`.
My only comment is about the version of the plugin, latest 1.0 seems more stable though 0.1 works also fine.

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3783 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3783/1/display/redirect

**Test results:**
- PASSED: 2951
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3783/1/testReport/org.jbpm.usertask.jpa.quarkus/PostgreSQLQuarkusJPAUserTaskInstancesTest/testFindByIdentityByPotentialGroups/">org.jbpm.usertask.jpa.quarkus.PostgreSQLQuarkusJPAUserTaskInstancesTest.testFindByIdentityByPotentialGroups</a></summary>
java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.testcontainers.quarkus.PostgreSqlQuarkusTestResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3783/1/testReport/org.kie.kogito.quarkus.workflows/ExpressionRestFromImageIT/testExpressionRest/">org.kie.kogito.quarkus.workflows.ExpressionRestFromImageIT.testExpressionRest</a></summary>
1 expectation failed.<br/>Expected status code <201> but was <404>.<br/>
</details>

**Comment by kie-ci3** [comment]
**PR job** `#2` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3783 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3783/2/display/redirect

**Test results:**
- PASSED: 3486
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3783/2/testReport/org.kie.kogito.quarkus.workflows/ExpressionRestFromImageIT/testExpressionRest/">org.kie.kogito.quarkus.workflows.ExpressionRestFromImageIT.testExpressionRest</a></summary>
1 expectation failed.<br/>Expected status code <201> but was <404>.<br/>
</details>

**Comment by kie-ci3** [comment]
**PR job** `#5` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3783 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3783/5/display/redirect

**Test results:**
- PASSED: 3480
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3783/5/testReport/org.kie.kogito.quarkus.workflows/ExpressionRestFromImageIT/testExpressionRest/">org.kie.kogito.quarkus.workflows.ExpressionRestFromImageIT.testExpressionRest</a></summary>
1 expectation failed.<br/>Expected status code <201> but was <404>.<br/>
</details>

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**